File: backend/app/schemas/vessel.py
from datetime import datetime
from pydantic import BaseModel, ConfigDict


class VesselPositionOut(BaseModel):
    model_config = ConfigDict(extra='allow')
    
    id: int | None = None
    mmsi: str
    ship_name: str | None = None
    imo: str | None = None
    callsign: str | None = None
    flag: str | None = None
    type: str | None = None
    latitude: float
    longitude: float
    speed: float | None = None
    course: float | None = None
    sog: float | None = None
    cog: float | None = None
    timestamp: datetime
    status: str | None = None
    suspicious: bool = False
    anomalyReason: str | None = None
    anomalySeverity: str | None = None

File: oceanshield-ai/src/types/vessel.ts
export type VesselType = 'Tanker' | 'Cargo' | 'Container' | 'Bulk Carrier' | 'Tug' | 'Fishing' | 'Passenger' | 'Other'

export type AnomalySeverity = 'LOW' | 'MEDIUM' | 'HIGH' | 'CRITICAL'

export interface AISPosition {
  latitude: number
  longitude: number
  timestamp: string
  speed?: number
  course?: number
  heading?: number
}

export interface Vessel {
  mmsi: string
  imo?: string
  name: string
  callsign?: string
  flag?: string
  type: VesselType
  length?: number
  beam?: number
  draft?: number
  latitude: number
  longitude: number
  speed?: number
  course?: number
  heading?: number
  timestamp: string
  status?: string
  suspicious: boolean
  anomalyReason?: string
  anomalySeverity?: AnomalySeverity
  destination?: string
  eta?: string
  routeHistory?: AISPosition[]
  aisGaps?: { startTime: string; endTime: string; durationMinutes: number }[]
}
